contributors/devel/sig-architecture/api_changes.md

```diff
@@ -154,18 +159,42 @@ without changing the API version, so you can simply change it to:
 // Still API v6.
 type Frobber struct {
   Height int    `json:"height"`
-  Width  int    `json:"width"`
+  Width  *int   `json:"width"`  // optional, defaults to 42
   Param  string `json:"param"`
 }
 ```
 
+Note that fields which are added MUST be optional.  Optional fields must be
+nil-able (pointers or slices or (rarely) maps).
```

Comment:
The second sentence doesn't follow from the first.

I agree with both sentences (albeit for different reasons than your motivation here), but I think the second one is an implementation detail, I'm not sure it deserves a "must".

Reply:
As I started this doc edit, I was overhauling the doc.  Then I realized that I would not finish in any reasonable time and dialed back the deltas.

I think we need to overhaul this doc totally, and as part of that we need to spec this (I do think it should be a MUST) as well as empty-list vs not-specified issues and other nuances that are poorly documented.

Reply:
I agree that this doc needs to be redone, I don't know how to find time to do it yet.

Also I basically think we cannot have this many rules for humans to follow, they need to be programmatic.